packages/grafana-ui/src/components/Input/Input.tsx

```diff
@@ -223,10 +237,10 @@ export const Input = React.forwardRef<HTMLInputElement, Props>((props, ref) => {
   const [suffixRect, suffixRef] = useClientRect<HTMLDivElement>();
 
   const theme = useTheme();
-  const styles = getInputStyles({ theme, invalid: !!invalid });
+  const styles = getInputStyles({ theme, invalid: !!invalid, width });
 
   return (
-    <div className={cx(styles.wrapper, inputSizes()[size], className)}>
+    <div className={cx(styles.wrapper, !(width || size === 'auto') && inputSizes()[size], className)}>
```

Comment:
and cannot say I understand the logic here. width is already applied in the getStyles wrapper. 

and should we not remove size here? it seems to do nothing we want. If we are to keep FormInputSize we should use ComponentSize and have it behave the same as Button.